src/Tools/dotnet-monitor/Egress/Extension/ExtensionEgressProvider.cs

```diff
@@ -0,0 +1,105 @@
+            : base(logger)
+        {
+            _propertyProvider = propertyProvider;
+        }
+
+        public override async Task<string> EgressAsync(
+            string providerType,
+            string providerName,
+            ExtensionEgressProviderOptions options,
+            Func<Stream, CancellationToken, Task> action,
+            EgressArtifactSettings artifactSettings,
+            CancellationToken token)
+        {
+            ExtensionEgressPayload payload = new ExtensionEgressPayload()
+            {
+                Settings = artifactSettings,
+                Configuration = options,
+                Properties = _propertyProvider.GetAllProperties(),
+                ProviderName = providerName,
+            };
+
+            /* [TODOs]
+             * 1. Add a new service to dynamically find these extension(s)
+             * 2. Remove all raw logging statements from this method and refactor into LoggingExtensions
+             * 3. Stream StdOut and StdErr async in the process so their streams don't need to end before we can return
+             * 4. Refactor WaitForExit to do an async wait
+             */
+            const string extensionProcessPath = "MyPathTo\\Extension.exe";
+            ProcessStartInfo pStart = new ProcessStartInfo()
+            {
+                FileName = extensionProcessPath,
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true,
+                UseShellExecute = false,
+            };
+            Process p = new Process()
+            {
+                StartInfo = pStart,
+            };
+
+            Logger?.LogInformation("Starting process...");
+            p.Start();
+            Logger?.LogInformation("starting stream...");
+
+            await JsonSerializer.SerializeAsync<ExtensionEgressPayload>(p.StandardInput.BaseStream, payload, options: null, token);
+            await p.StandardInput.WriteLineAsync();
+            await p.StandardInput.BaseStream.FlushAsync(token);
+
+            await action(p.StandardInput.BaseStream, token);
+            await p.StandardInput.BaseStream.FlushAsync(token);
+
+            Logger?.LogInformation("Stream done, sending close...");
+
+            p.StandardInput.Close();
+
+            Logger?.LogInformation("Waiting for exit...");
+
+            string procStdOutContent = await p.StandardOutput.ReadToEndAsync();
+            string procErrorContent = await p.StandardError.ReadToEndAsync();
+
+            await p.WaitForExitAsync(token);
+            Logger?.LogInformation($"Exited with code: {p.ExitCode}");
+
+            if (!string.IsNullOrWhiteSpace(procErrorContent))
+            {
+                Logger?.LogWarning($"Process Error output: {procErrorContent}");
+            }
+
+            return procStdOutContent;
```

Comment:
Please add a TODO comment that this need to be the egressed path. You might have to devise a standards-based protocol for stdout to report the path and a success value, and possibly logging at various levels too.